prime-router/src/main/kotlin/transport/SftpTransport.kt

```diff
@@ -67,33 +71,34 @@ class SftpTransport : ITransport {
     }
 
     companion object {
-        fun lookupCredentials(receiverFullName: String): Pair<String, String> {
+        fun lookupCredentials(receiverFullName: String): SftpCredential {
             val credentialLabel = receiverFullName
                 .replace(".", "--")
                 .replace("_", "-")
                 .toUpperCase()
 
-            // Assumes credential will be cast as UserPassCredential, if not return null, and thus the error case
-            val credential = CredentialHelper.getCredentialService().fetchCredential(
+            // Assumes credential will be cast as SftpCredential, if not return null, and thus the error case
+            return CredentialHelper.getCredentialService().fetchCredential(
                 credentialLabel, "SftpTransport", CredentialRequestReason.SFTP_UPLOAD
-            ) as? UserPassCredential?
+            ) as? SftpCredential?
                 ?: error("Unable to find SFTP credentials for $receiverFullName connectionId($credentialLabel)")
-
-            return Pair(credential.user, credential.pass)
         }
 
         fun connect(
             host: String,
             port: String,
-            user: String,
-            pass: String,
+            credential: SftpCredential,
         ): SSHClient {
             // create our client
             val sshClient = SSHClient()
             try {
                 sshClient.addHostKeyVerifier(PromiscuousVerifier())
                 sshClient.connect(host, port.toInt())
-                sshClient.authPassword(user, pass)
+                when (credential) {
+                    is UserPassCredential -> sshClient.authPassword(credential.user, credential.pass)
+                    is UserPpkCredential -> PuTTYKeyFile().init(StringReader(credential.key))
```

Comment:
In this case, how does the ppk file get associated with the sshClient?

Reply:
Fixed